File: src/fr/simoncros/computorv1/lexer/Token.java
package fr.simoncros.computorv1.lexer;

import java.text.NumberFormat;
import java.util.Locale;

public class Token {
    public TokenType type;
    public String identifier = null;
    public float number = 0;

    public Token(TokenType type) {
        this.type = type;
    }

    public static Token newIdentifier(String value) {
        Token token = new Token(TokenType.IDENTIFIER);
        token.identifier = value;
        return token;
    }

    public static Token newNumber(float value) {
        Token token = new Token(TokenType.NUMBER);
        token.number = value;
        return token;
    }

    @Override
    public String toString() {
        return switch (this.type) {
            case ADD -> "+";
            case SUB -> "-";
            case MUL -> "*";
            case DIV -> "/";
            case POW -> "^";
            case EQUAL -> "=";
            case LPARENT -> "(";
            case RPARENT -> ")";
            case SPACE -> " ";
            case IDENTIFIER -> this.identifier;
            case NUMBER -> NumberFormat.getNumberInstance(Locale.ENGLISH).format(number);
            case EOF -> "";
        };
    }
}

File: rust/src/lexer/token.rs
use std::fmt::Debug;

pub enum Token {
    Add(char),
    Sub(char),
    Div(char),
    Mul(char),
    Pow(char),
    Equal(char),
    LParent(char),
    RParent(char),
    Identifier(String),
    Number(f32),
    EOF,
}

impl Debug for Token {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        match self {
            Self::Add(v) | Self::Sub(v) | Self::Mul(v) | Self::Div(v) | Self::Pow(v) | Self::Equal(v) | Self::LParent(v) | Self::RParent(v) => write!(f, "{:?}", v),
            Self::Identifier(v) => write!(f, "{:?}", v),
            Self::Number(v) => write!(f, "{:?}", v),
            Self::EOF => write!(f, ""),
        }
    }
}
